src/archetype.rs

```diff
@@ -285,22 +285,13 @@ impl Archetype {
                             )
                             .unwrap(),
                         );
+                        let mem = NonNull::new(mem).unwrap();
```

Comment:
Docs suggest using `handle_alloc_error` instead of panicking directly, but no big deal.

Reply:
Didn't know about that, thanks! https://github.com/Ralith/hecs/pull/350.